addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -395,3 +395,95 @@ def test_multicompany_partner_bank(self):
 
         self.assertEqual(po_b.invoice_ids.company_id, company_b)
         self.assertEqual(po_b.invoice_ids.partner_bank_id, partner_bank_b)
+
+    def test_sequence_invoice_lines_from_multiple_purchases(self):
+        """Test if the invoice lines are sequenced by purchase order when creating an invoice
+           from multiple selected po's"""
+        purchase_order_ids = []
+        PurchaseOrderLine = self.env['purchase.order.line'].with_context(tracking_disable=True)
+
+        for sequence_number in range(3):
+            purchase_order = self.env['purchase.order'].with_context(tracking_disable=True).create({
+                'partner_id': self.partner_a.id,
+            })
+            for sequence_number in range(10, 13):
+                vals_pol = {
+                    'name': self.product_order.name,
+                    'product_id': self.product_order.id,
+                    'product_qty': 10.0,
+                    'product_uom': self.product_order.uom_id.id,
+                    'price_unit': self.product_order.list_price,
+                    'order_id': purchase_order.id,
+                    'taxes_id': False,
+                    'sequence': sequence_number,
+                }
+                PurchaseOrderLine.create(vals_pol)
+            purchase_order.button_confirm()
+            purchase_order_ids.append(purchase_order.id)
+
+        action = self.env['purchase.order'].browse(purchase_order_ids).action_create_invoice()
+        invoice = self.env['account.move'].browse(action['res_id'])
+
+        expected_sequence_purchase_id = {
+            10:  purchase_order_ids[0],
+            11:  purchase_order_ids[0],
+            12:  purchase_order_ids[0],
+            13:  purchase_order_ids[1],
+            14:  purchase_order_ids[1],
+            15:  purchase_order_ids[1],
+            16:  purchase_order_ids[2],
+            17:  purchase_order_ids[2],
+            18:  purchase_order_ids[2],
+        }
+
+        for line in invoice.line_ids.filtered(lambda l: l.exclude_from_invoice_tab is False):
+            expected_purchase_id = expected_sequence_purchase_id[line.sequence]
+            self.assertEqual(line.purchase_order_id.id, expected_purchase_id)
```

Comment:
I think having the sequence hardcoded isn't relevant, only the PO lines' order is.
I suggest to only define the PO order and check that for each invoice line
```suggestion
         expected_purchase = [
            purchase_orders[0], purchase_orders[0], purchase_orders[0],
            purchase_orders[1], purchase_orders[1], purchase_orders[1],
            purchase_orders[2], purchase_orders[2], purchase_orders[2],
        ]

        for line in invoice.invoice_line_ids:
            self.assertEqual(line.purchase_order_id, expected_purchase.pop(0))
```
Also, you can use `invoice_line_ids` instead of filter the `line_ids`